components/script/dom/treewalker.rs

```diff
@@ -417,21 +420,33 @@ impl TreeWalker {
     // https://dom.spec.whatwg.org/#concept-node-filter
     fn accept_node(&self, node: &Node) -> Fallible<u16> {
         // "To filter node run these steps:"
-        // "1. Let n be node's nodeType attribute value minus 1."
+        // "1. If the active flag is set, then throw an InvalidStateError."
+        if self.active.get() {
+            return Err(Error::InvalidState);
+        }
+        // "2. Let n be node's nodeType attribute value minus 1."
         let n = node.NodeType() - 1;
-        // "2. If the nth bit (where 0 is the least significant bit) of whatToShow is not set,
+        // "3. If the nth bit (where 0 is the least significant bit) of whatToShow is not set,
         //     return FILTER_SKIP."
         if (self.what_to_show & (1 << n)) == 0 {
             return Ok(NodeFilterConstants::FILTER_SKIP)
         }
-        // "3. If filter is null, return FILTER_ACCEPT."
-        // "4. Let result be the return value of invoking filter."
-        // "5. If an exception was thrown, re-throw the exception."
-        // "6. Return result."
+        // "4. If filter is null, return FILTER_ACCEPT."
+        // "5. Set the active flag."
+        // "6. Let result be the return value of call a user object’s operation with filter,
+        //     "acceptNode", and a list of arguments consisting of node. If that throws an exception,
+        //     then unset the active flag and rethrow the exception."
+        // "7. Unset the active flag"
+        // "8. Return result."
```

Comment:
I'm not sure why these step annotations are there, we usually just do `step n`.

Reply:
@KiChjang Due to the existing step annotation, I just updated them to the new one. If they're unnecessary, please let me remove them.

Reply:
These seem to be left by a contributor back in #3253, when we did not have guidelines as to how we're supposed to leave step annotations.
So yes, please change them back into `step n` instead.